### PR TITLE
Fix error message of statefulset test

### LINKED ISSUE #50592: Unit test error message should follow go style
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind feature

**What happened**:
Unit test error message match the conventional Go style:

https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures
For example:

if got, want := set.Status.Replicas, 3; got != want {
  t.Errorf("set.Status.Replicas = %v; want %v", got, want)
}

### PR DESCRIPTION
**Pull request by zhangxiaoyu-zidif**
**What this PR does / why we need it**:
Fix error message of statefulset test
It should be 0 replocas in the error message.
And fix typo from Falied to Failed

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes https://github.com/kubernetes/kubernetes/issues/50592

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/statefulset/stateful_set_test.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func TestStatefulSetControllerDeletes(t *testing.T) {
 		set = obj.(*apps.StatefulSet)
 	}
 	if set.Status.Replicas != 0 {
-		t.Error("Falied to scale statefulset to 3 replicas")
+		t.Error("Falied to scale statefulset to 0 replicas")
```
Comment (xiangpengzhao) [inline pkg/controller/statefulset/stateful_set_test.go]: s/Falied/Failed

**Review by xiangpengzhao — CHANGES_REQUESTED** [review]

File: pkg/controller/statefulset/stateful_set_test.go (hunk 1 above)
Comment (zhangxiaoyu-zidif) [inline pkg/controller/statefulset/stateful_set_test.go]: Thanks for that. =)

File: pkg/controller/statefulset/stateful_set_test.go [hunk 2]
```diff
@@ -125,7 +125,7 @@ func TestStatefulSetControllerRespectsTermination(t *testing.T) {
 		set = obj.(*apps.StatefulSet)
 	}
 	if set.Status.Replicas != 0 {
-		t.Error("Falied to scale statefulset to 3 replicas")
+		t.Error("Failed to scale statefulset to 3 replicas")
```
Comment (xiangpengzhao) [inline pkg/controller/statefulset/stateful_set_test.go]: s/3/0

**Review by xiangpengzhao — CHANGES_REQUESTED** [review]
just one nit, otherwise LGTM.

File: pkg/controller/statefulset/stateful_set_test.go [hunk 3]
```diff
@@ -46,7 +46,7 @@ func TestStatefulSetControllerCreates(t *testing.T) {
 		set = obj.(*apps.StatefulSet)
 	}
 	if set.Status.Replicas != 3 {
-		t.Error("Falied to scale statefulset to 3 replicas")
+		t.Error("Failed to scale statefulset to 3 replicas")
```
Comment (enisoc) [inline pkg/controller/statefulset/stateful_set_test.go]: Ideally these should be changed to match the conventional Go style:

https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures

```go
if got, want := set.Status.Replicas, 3; got != want {
  t.Errorf("set.Status.Replicas = %v; want %v", got, want)
}
```

Would you be willing to update the PR to that form?

Comment (zhangxiaoyu-zidif) [inline pkg/controller/statefulset/stateful_set_test.go]: yes, my pleasure. =) I will do it~

**Comment by zhangxiaoyu-zidif** [comment]
@enisoc @xialonglee  PTAL

**Comment by zhangxiaoyu-zidif** [comment]
@enisoc @xiangpengzhao 
PTAL

**Pull request merged**